controllers/front/callbackfail.php

```diff
@@ -128,29 +132,39 @@ public function postProcess()
                 // Redirect to payment selection page
                 if ($fraudPayment['payment_status']) {
                     saveLogs($transaction->FraudExplanation);
-                    redirectUserToCheckoutPaymentStep($lockFileName, $lockFileHandle);
+                    !$isChildOrder ? redirectUserToCheckoutPaymentStep($lockFileName, $lockFileHandle) : exit('Error in the Payment!');
                 } else {
                     // Check if an order exist
-                    $order = getOrderFromUniqueId($shopOrderId);
+                    $order = $isChildOrder ? getChildOrderFromUniqueId($shopOrderId) : getOrderFromUniqueId($shopOrderId);
+
                     if (Validate::isLoadedObject($order)) {
-                        updateOrder($cart, $order, $response, $shopOrderId, $lockFileName, $lockFileHandle);
-                    } else {
+                        if ($isChildOrder) {
+                            updateChildOrder($cart, $order, $response, $shopOrderId, $lockFileName, $lockFileHandle);
+                        } else {
+                            updateOrder($cart, $order, $response, $shopOrderId, $lockFileName, $lockFileHandle);
+                        }
+                    } elseif (!$isChildOrder) {
                         createOrder($response, $currencyPaid, $cart, $orderStatus);
                     }
                 }
                 // Load order
-                $order = new Order((int) $this->module->currentOrder);
+                if ($isChildOrder) {
+                    $order_id = Order::getOrderByCartId((int) ($cart->id));
+                    $order = new Order((int) $order_id);
```

Comment:
Same as above

Reply:
@emicha, we make sure the cart object and its ID are valid before using them in the code. Here's how we do it:

```
            // Load the cart
            $cart = getCartFromUniqueId($shopOrderId);
            if (!Validate::isLoadedObject($cart)) {
                exit('Could not load cart - exiting');
            }
```
This check ensures that cart->id is always valid, which prevents any potential issues with casting.